rust/src/main.rs

```diff
@@ -0,0 +1,272 @@
+//! proto files for further compilation. This is based on the proto-compiler code
+//! in github.com/informalsystems/ibc-rs
+
+use regex::Regex;
+use std::{
+    env,
+    ffi::OsStr,
+    fs::{self, create_dir_all, remove_dir_all},
+    io,
+    path::{Path, PathBuf},
+    sync::atomic::{self, AtomicBool},
+};
+use walkdir::WalkDir;
+
+/// Suppress log messages
+// TODO(tarcieri): use a logger for this
+static QUIET: AtomicBool = AtomicBool::new(false);
+
+
+// All paths must end with a / and either be absolute or include a ./ to reference the current
+// working directory.
+
+/// The directory generated cosmos-sdk proto files go into in this repo
+const OUTPUT_DIR: &str = "./";
+/// Directory where the cosmos-sdk proto are located
+const COSMOS_SDK_DIR: &str = "../proto/cosmos-sdk";
+/// Directory where the okp4 proto are located
+const OKP4_DIR: &str = "../proto/okp4";
+
+/// A temporary directory for proto building
+const TMP_BUILD_DIR: &str = "/tmp/tmp-protobuf/";
+
+// Patch strings used by `copy_and_patch`
+
+/// Protos belonging to these Protobuf packages will be excluded
+/// (i.e. because they are sourced from `tendermint-proto`)
+const EXCLUDED_PROTO_PACKAGES_COSMOS: &[&str] = &["gogoproto", "google", "tendermint"];
+const EXCLUDED_PROTO_PACKAGES_OKP4: &[&str] = &["gogoproto", "google", "tendermint", "cosmos", "cosmos_proto"];
+
+/// Log info to the console (if `QUIET` is disabled)
+// TODO(tarcieri): use a logger for this
+macro_rules! info {
+    ($msg:expr) => {
+        if !is_quiet() {
+            println!("[info] {}", $msg)
+        }
+    };
+    ($fmt:expr, $($arg:tt)+) => {
+        info!(&format!($fmt, $($arg)+))
+    };
+}
+
+fn main() {
+    if is_github() {
+        set_quiet();
+    }
+
+    let tmp_build_dir: PathBuf = TMP_BUILD_DIR.parse().unwrap();
+    let proto_dir: PathBuf = OUTPUT_DIR.parse().unwrap();
+
+    if tmp_build_dir.exists() {
+        fs::remove_dir_all(tmp_build_dir.clone()).unwrap();
+    }
+
+    let temp_sdk_dir = tmp_build_dir.join("cosmos-sdk");
+    let temp_okp4_dir = tmp_build_dir.join("okp4");
+
+    fs::create_dir_all(&temp_sdk_dir).unwrap();
+    fs::create_dir_all(&temp_okp4_dir).unwrap();
+
+    compile_sdk_protos_and_services(&temp_sdk_dir);
+    compile_okp4_protos_and_services(&temp_okp4_dir);
+
+    copy_generated_files(&temp_sdk_dir, &proto_dir.join("cosmos_sdk_grpc_client").join("src"), EXCLUDED_PROTO_PACKAGES_COSMOS);
+    copy_generated_files(&temp_okp4_dir, &proto_dir.join("okp4_grpc_client").join("src"), EXCLUDED_PROTO_PACKAGES_OKP4);
+
+}
+
+fn is_quiet() -> bool {
+    QUIET.load(atomic::Ordering::Relaxed)
+}
+
+fn set_quiet() {
+    QUIET.store(true, atomic::Ordering::Relaxed);
+}
+
+/// Parse `--github` flag passed to `proto-build` on the eponymous GitHub Actions job.
+/// Disables `info`-level log messages, instead outputting only a commit message.
+fn is_github() -> bool {
+    env::args().any(|arg| arg == "--github")
```

Comment:
This args seem to not be used into the CI, maybe you can remove this unecessary condition.